app/controllers/general_shopping_lists_controller.rb

```diff
@@ -0,0 +1,14 @@
+class GeneralShoppingListsController < ApplicationController
+  def index
+    @recipe_foods = Food.joins(
+      "INNER JOIN (SELECT food_id,
+                                SUM(quantity) as total_quantity,
+                                SUM(value) as total_value
+                         FROM recipe_foods GROUP BY food_id) as subquery
+               ON foods.id = subquery.food_id"
+    )
+      .select('foods.name,
+                       subquery.total_quantity,
+                       subquery.total_value')
```

Comment:
You did a great job! but Your current approach of placing a complex SQL query directly in the controller might work. Still, it is generally advisable to move such logic into the models for better organization, maintainability, and adherence to the MVC (Model-View-Controller) architecture. Placing SQL queries in the models also promotes reusability and makes it easier to test and modify the queries. Please update your code 🙏 .

Example:
1. Assuming you have models for Food and RecipeFood, you could define a method in the Food model to encapsulate this logic then in the controller you can simply call that method like below:
```
# app/models/food.rb

class Food < ApplicationRecord
  def self.recipe_totals
    joins(
      "INNER JOIN (SELECT food_id,
                                SUM(quantity) as total_quantity,
                                SUM(value) as total_value
                         FROM recipe_foods GROUP BY food_id) as subquery
               ON foods.id = subquery.food_id"
    )
    .select('foods.name, subquery.total_quantity, subquery.total_value')
  end
end
```
```
# app/controllers/general_shopping_lists_controller.rb
class GeneralShoppingListsController < ApplicationController
  def index
    @recipe_foods = Food.recipe_totals
  end
end
```